appoetry1/Flows/AppFlow.swift

```diff
@@ -9,13 +9,13 @@
 import UIKit
 
 protocol FlowController: class {
-    var window: UIWindow?? { get }
+    var window: UIWindow { get }
     func start()
 }
 
 extension FlowController {
-    var window: UIWindow?? {
-        return UIApplication.shared.delegate!.window
+    var window: UIWindow {
+        return (UIApplication.shared.delegate!.window ?? UIWindow()) ?? UIWindow()
```

Comment:
Better just leave the 'window' property as optional in AppDelegate.

In this var return UIApplication.shared.delegate?.window! and have the return type as UIWindow?

So, in code you can address the optional window without the double optional mess. 

There is no need for ?? window ?? window, because they will not exist anyway, they are not a safe way out